compiler/rustc_middle/src/ty/mod.rs

```diff
@@ -2410,6 +2410,23 @@ impl<'tcx> TyCtxt<'tcx> {
         }
     }
 
+    pub fn get_attrs_by_path(
+        self,
+        did: impl Into<DefId>,
```

Comment:
Why does this take an `Into<DefId>`? It's really general -- please just take a `DefId`, and make the caller call `into_def_id()` if needed.